### PR TITLE
Use flake8-force plugin for flake8

### PR DESCRIPTION
**Pull request by tommy-waltmann**
## Description
This PR has the necessary changes to use the flake8-force plugin for flake8. 

## Motivation and Context
The 4.0 release of flake8 made it unable to lint cython files (see #897 #884 ). A new plugin now exists to allow the old flake8 behavior and this PR adds the new plugin to freud's pre-commit configuration and hooks.

## How Has This Been Tested?
flake8 hook should work in the CI and when pre-commit is run locally.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds or improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation improvement (updates to user guides, docstrings, or developer docs)

## Checklist:

- [ ] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/freud/blob/master/CONTRIBUTING.md) document.
- [ ] My code follows the code style of this project.
- [ ] I have updated the documentation (if relevant).
- [ ] I have added tests that cover my changes (if relevant).
- [ ] All new and existing tests passed.
- [ ] I have updated the [credits](https://github.com/glotzerlab/freud/blob/master/doc/source/reference/credits.rst).
- [ ] I have updated the [Changelog](https://github.com/glotzerlab/freud/blob/master/ChangeLog.md).

### CODE REVIEW

**Comment by tommy-waltmann** [comment]
Everything appears to be in working order now.

File: freud/order.pyx [hunk 1]
```diff
@@ -649,7 +651,7 @@ cdef class Steinhardt(_PairCompute):
         computed by averaging the :math:`q_{lm}` values (or
         :math:`\overline{q}_{lm}` values if ``average`` is enabled) over all
         particles before computing the rotationally-invariant order
-        parameter."""
+        parameter."""  # noqa W605
```
Comment (bdice) [inline freud/order.pyx]: This probably needs to be a raw string `r"""` instead. This warning is indicating that `\o` is not a valid escape sequence.

**Review by bdice — CHANGES_REQUESTED** [review]

File: freud/locality.pyx [hunk 2]
```diff
@@ -424,7 +427,8 @@ cdef class NeighborQuery:
                 :meth:`matplotlib.axes.Axes.plot`.
 
         Returns:
-            tuple (:class:`matplotlib.axes.Axes`, :class:`matplotlib.collections.PathCollection`):
+            tuple (:class:`matplotlib.axes.Axes`,
+                   :class:`matplotlib.collections.PathCollection`):
```
Comment (bdice) [inline freud/locality.pyx]: Sphinx doesn't like this. I'm not sure exactly what it needs. 
image

**Review by bdice — COMMENTED** [review]
Small issue that could be fixed in this PR: `freud.diffraction.DiffractionPattern.k_vectors` docs aren't rendering right:
image

It should have a "**Type:**" block like this example, `freud.diffraction.StaticStructureFactorDirect.S_k`:
image

**Comment by tommy-waltmann** [comment]
> Small issue that could be fixed in this PR: `freud.diffraction.DiffractionPattern.k_vectors` docs aren't rendering right: image
> 
> It should have a "**Type:**" block like this example, `freud.diffraction.StaticStructureFactorDirect.S_k`: image

The `k_vectors` not rendering right has something to do with the fact that there are two things enclosed with `` in the docstring, not just one. I have tried playing with this and have not had success.

EDIT: In `DiffractionPattern.to_image` we return a similar tuple but do not enclose output_size in \``, so another option is to just remove the \`` formatting

**Review by bdice — APPROVED** [review]

**Comment by bdice** [comment]
@tommy-waltmann The issue wasn't the backticks -- it was that the line break fell right at the end of the type specification, which messed up Sphinx's parsing. I escaped the newline to fix this.

**Pull request merged**